avm/src/daemon/ctrl.rs

```diff
@@ -705,6 +705,32 @@ impl ControlPort {
     }
   }
 
+  pub async fn is_cluster_up(self: &mut ControlPort) -> bool {
+    let cluster_secret = CLUSTER_SECRET.get().unwrap().clone().unwrap();
+    let mut health = vec![];
+    let nodes = self.lrh.get_mut_nodes();
+    for node in nodes.iter() {
+      let mut req = Request::builder()
+        .method("GET")
+        .uri(format!("https://{}:4142/clusterHealth", node.id));
+      req = req.header(cluster_secret.as_str(), "true");
+      health.push(self.client.request(req.body(Body::empty()).unwrap()));
+    }
+    let health_res = join_all(health).await;
+    let mut results = Vec::new();
+    for res in health_res.iter() {
+      match res {
+        Err(_) => {
+          results.push(false);
+        }
+        Ok(res) => {
+          results.push(res.status().as_u16() == 200);
+        }
+      }
+    }
+    return results.iter().all(|e| *e);
+  }
```

Comment:
This is kinda going in the opposite direction that we're trying to go. The cluster should be able to use its own state tracking to determine if it's up or down instead of asking the deploy service this. Why can't you use the `self.vms_up` HashMap and just return true if all of the booleans are true?

Reply:
I'm not asking to deploy service, I'm just checking `clusterHealth` through control port. I am not using `self.vms_up` since that vector is filled within the loop that I'm short circuiting to avoid unnecessary leader and `/stats` logic. But I guess I could add my check later after the `update_vms`method is called to use the `vms_up`.
https://github.com/alantech/alan/pull/567/files#diff-583f2380a4d780cb04bd93e3492a6f2828052a50e23460f3a2c8d89cf3ade79fR303-R311

Reply:
Oh, sorry. Still waking up. I guess I didn't even consider that since the `vms_up` hashmap already exists and does roughly the same thing.

But if you short-circuit out the code that fills that, you'll break other parts of the daemon (how it knows which nodes can have data written to it, or which nodes should be included in the load balancer, etc).

Reply:
Oh yes I might break some of that logic, let me move this check then